platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyServices.java

```diff
@@ -43,12 +43,12 @@ public void registerProjectServices(ServiceRegistration registration) {
 
     @Override
     public void registerBuildSessionServices(ServiceRegistration registration) {
-        registration.add(ArtifactSetToFileCollectionFactory.class);
         registration.addProvider(new DependencyManagementBuildSessionScopeServices());
     }
 
     @Override
     public void registerBuildTreeServices(ServiceRegistration registration) {
+        registration.add(ArtifactSetToFileCollectionFactory.class);
```

Comment:
This needs to be BuildTree-scoped to inject into `ArtifactSetToFileCollectionFactory`.  Moving a file collection factory from BuildSession down to BuildTree scope seems harmless enough.